### PR TITLE
Add Pushgateway 0.4.0 for Prometheus

### PR DESCRIPTION
**Pull request by gongled**
Hello,

In addition to PR #182 I have prepared Prometheus Pushgateway RPM spec. It provides mechanism to push custom metrics from any applications for being scraped from Prometheus.

### CODE REVIEW

File: prometheus-pushgateway/prometheus-pushgateway.spec [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+fi
+%endif
+
+################################################################################
+
+%files
+%defattr(-,root,root,-)
+%{_bindir}/%{name}
+%if 0%{?rhel} >= 7
+    %{_unitdir}/%{name}.service
```
Comment (andyone) [inline prometheus-pushgateway/prometheus-pushgateway.spec]: Don't use prefix spaces in `%files` section.

File: prometheus-pushgateway/prometheus-pushgateway.spec [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+%clean
+rm -rf %{buildroot}
+
+################################################################################
+
+%pre
+%{__getent} group %{service_group} >/dev/null || %{__groupadd} -r %{service_group}
+%{__getent} passwd %{service_user} >/dev/null || %{__useradd} -r \
+    -g %{service_group} -d %{_sharedstatedir}/%{name} \
```
Comment (andyone) [inline prometheus-pushgateway/prometheus-pushgateway.spec]: You have the same user in `prometheus` package, but in this package, you try to create the user with a different home directory.

**Review by andyone — CHANGES_REQUESTED** [review]

File: prometheus-pushgateway/prometheus-pushgateway.spec (hunk 2 above)
Comment (gongled) [reply prometheus-pushgateway/prometheus-pushgateway.spec]: Oops, my bad. It seems that `/var/lib/prometheus` will never exist whether user install only Pushgateway. So I decided to change the username, group and home directory. (b845430e2f66ba0f0ff1ef2e9cae99a577b80c4b)

File: prometheus-pushgateway/prometheus-pushgateway.spec (hunk 1 above)
Comment (gongled) [reply prometheus-pushgateway/prometheus-pushgateway.spec]: Fixed (b11385b0389466a952508b7cf1e0c755d1f8add7).

File: prometheus-pushgateway/SOURCES/prometheus-pushgateway.init [hunk 3]
```diff
@@ -29,7 +29,7 @@ kv[prog_name]="prometheus-pushgateway"
 kv.readSysconfig
 
 binary="${BINARY:-/usr/bin/prometheus-pushgateway}"
-user="${USER:-prometheus}"
+user="${USER:-prometheus-pushgateway}"
```
Comment (andyone) [inline prometheus-pushgateway/SOURCES/prometheus-pushgateway.init]: I think it will be much better if a user will be just `pushgateway`, don't you think?

Comment (gongled) [reply prometheus-pushgateway/SOURCES/prometheus-pushgateway.init]: Yes. This long-length name looks awful. Use `pushgateway` instead (c9d9525a6457dbbe6083cecc53da455ba9df145d).

**Review by andyone — APPROVED** [review]

**Comment by andyone** [comment]
Looks good to me.

**Comment by andyone** [comment]
The package was added to `kaos-testing` repository.

**Pull request merged**